mail_forward/models/compose_message.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# © 2014-2015 Grupo ESOC <www.grupoesoc.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class MailForwardComposeMessage(models.TransientModel):
+    """Allow forwarding a message.
+
+    It duplicates the message and optionally attaches it to another object
+    of the database and sends it to another recipients than the original one.
+    """
+    _name = "mail_forward.compose_message"
+    _inherits = {"mail.compose.message": "original_wizard_id"}
+
+    @api.model
+    def default_get(self, fields):
+        """Fix default values.
+
+        Sometimes :meth:`openerp.addons.mail.mail_compose_message
+        .mail_compose_message.default_get` overwrites the default value
+        for the ``subject`` field, even when it gets the right default value
+        from the context.
+
+        This method fixes that by getting it from the context if available.
+        """
+        result = self.original_wizard_id.default_get(fields)
+
+        if "subject" in result and "default_subject" in self.env.context:
+            result["subject"] = self.env.context["default_subject"]
+
+        return result
+
+    @api.model
+    def _get_model_selection(self):
+        """Get allowed models and their names."""
+        model_objs = self.env["res.request.link"].search(
+            [("mail_forward_target", "=", True)],
+            order="name")
+        return [(m.object, m.name) for m in model_objs]
+
+    @api.onchange("destination_object_id")
+    def change_destination_object(self):
+        """Update some fields for the new message."""
+        if self.destination_object_id:
+            self.model = self.destination_object_id._name
+            self.res_id = self.destination_object_id.id
+
+            model_name = (self.env["ir.model"]
+                          .search([("model", "=", self.model)])
+                          .name)
+            record_name = self.destination_object_id.name_get()[0][1]
+            if model_name:
+                record_name = "%s %s" % (model_name, record_name)
+
+            self.record_name = record_name
+
+    @api.multi
+    def send_mail_action(self):
+        return self.send_mail()
```

Comment:
Why to declare 2 methods to do the same? Am I missing something? :thinking: Otherwise, remove one of them please. :wink:

Reply:
The way odoo mail_compose_message is done works the same, if we don't declare them it will fail. I tried to remove this one but then it fails.

Reply:
OK, I see now. However, [you are repeating here the same code as upstream](https://github.com/odoo/odoo/blob/33b698254a00862991d8596b72c3e3b45ebbfbf3/addons/mail/wizard/mail_compose_message.py#L186-L194), so it still makes no sense that it fails... (You should only override a method if anything changes)

Besides, it breaks inheritance too. In case this were needed, it should be:

```python
return super(MailComposeMessage, self).send_mail()
```

Could you double-check this please?

Reply:
Oh snap, excuse me! It was right as it was. I didn't realize this was a `_inherits` model, not an `_inherit` one.

This kind of inheritance inherits fields, but not methods. Possibly calling `super()` will not work, and the correct code is the one you had indeed.

Just please add a comment similar to upstream's, so reviewers can understand the reasoning behind that code.

Thanks, and sorry!

Reply:
No worries, I didn't know as well that `_inherits` don't inherit the methods. Now it's clear why it's needed. I added the comment and restore the previous version. Thanks for your review.